src/vivarium_public_health/disease/state.py

```diff
@@ -212,13 +211,30 @@ def setup(self, builder):
 
         if isinstance(self._dwell_time, pd.DataFrame) or self._dwell_time.days > 0:
             self.transition_set.allow_null_transition = True
-            self.track_events = True
 
         if isinstance(self._dwell_time, pd.Timedelta):
             self._dwell_time = self._dwell_time.total_seconds() / (60*60*24)
         self.dwell_time = builder.value.register_value_producer(f'{self.state_id}.dwell_time',
                                                                 source=builder.lookup.build_table(self._dwell_time))
 
+    def load_population_columns(self, pop_data):
+        super().load_population_columns(pop_data)
+        simulants_with_condition = self.population_view.get(pop_data.index, query=f'{self._model}=="{self.state_id}"')
+
+        if not simulants_with_condition.empty and np.any(self.dwell_time(simulants_with_condition)) > 0:
+            infected_at = self._assign_event_time_for_prevalent_cases(simulants_with_condition, self.clock(),
+                                                                      self.randomness.get_draw, self.dwell_time)
+            infected_at.name = self.event_time_column
+            self.population_view.update(infected_at)
+
+    @staticmethod
+    def _assign_event_time_for_prevalent_cases(infected, current_time, randomness_func, dwell_time_func):
+        if not np.all(dwell_time_func(infected.index)) > 0:
```

Comment:
I also thought that it's  strange to have different dwell times for different simulants. But as you said, it's technically possible especially because they can always provide the custom data and we don't know what's in there. When I talked to James, he thinks that we may raise an error for this weird cases (probably means that modelers trying to do something weird?)  I honestly don't know :( we can just make it a warning and assign the current time as you wrote.

Reply:
I thought the conclusion we came to is that modelers will either provide a single number (e.g. acute mi which lasts 28 days) or we'd be converting from remission rates and so we should never have an issue.